### PR TITLE
Move tests into their own folder + general cleanup

### PR DESCRIPTION
**Pull request by ian-axelrod**
Having tests in a folder separate from the main library (oz) is standard
practice for python libraries, and lets us easily exclude test code from
the distribution.

### CODE REVIEW

File: oz/error_pages/__init__.py [hunk 1]
```diff
@@ -41,7 +40,8 @@ def get_lines_from_file(filename, lineno, context_lines):
     Returns (pre_context_lineno, pre_context, context_line, post_context).
     """
 
-    get_lines = lambda s, e: [linecache.getline(filename, l).rstrip() for l in range(s, e)]
```
Comment (ysimonson) [inline oz/error_pages/__init__.py]: Could you undo this change? I think the lambda is fine to use here.

Comment (chrisjryan) [inline oz/error_pages/__init__.py]: Actually, lambda expressions in cases like this are not preferred by PEP8:

["Always use a def statement instead of an assignment statement that binds a lambda expression directly to an identifier."](https://www.python.org/dev/peps/pep-0008/#programming-recommendations)

Comment (ysimonson) [inline oz/error_pages/__init__.py]: Yeah, we just had this same discussion in a PR for one of our internal projects, and the conclusion came down to "we use lambda pervasively, it hasn't been a problem, so it's fine to go against pep8 on this one like we do on occasion." Though thinking about it, I suppose that logic doesn't necessarily apply to our open source code. We can keep this change.
